src/tests/suite/daos_obj.c

```diff
@@ -523,6 +523,69 @@ lookup_empty_single(const char *dkey, const char *akey, uint64_t idx,
 	       req, true);
 }
 
+/**
+ * get the Pool storage info.
+ */
+static int
+pool_storage_info(void **state, daos_pool_info_t *pinfo)
+{
+	test_arg_t *arg = *state;
+	int rc;
+
+	/*get only pool space info*/
+	pinfo->pi_bits = DPI_SPACE;
+	rc = daos_pool_query(arg->pool.poh, NULL, pinfo, NULL, NULL);
+	if (rc != 0) {
+		print_message("pool query failed %d\n", rc);
+		return rc;
+	}
+
+	print_message("SCM space: Total = %" PRIu64 " Free= %" PRIu64"\t"
+	"NVMe space: Total = %" PRIu64 " Free= %" PRIu64"\n",
+	pinfo->pi_space.ps_space.s_total[0],
+	pinfo->pi_space.ps_space.s_free[0],
+	pinfo->pi_space.ps_space.s_total[1],
+	pinfo->pi_space.ps_space.s_free[1]);
+
+	return rc;
+}
+
+/**
+ * Disabled Aggrgation for Pool.
+ */
+static int
+disable_pool_aggregation(void **state, daos_pool_info_t *pinfo)
+{
+	test_arg_t *arg = *state;
+	int rc;
+	char const *const names[] = {"reclaim"};
+	void const *const in_values[] = {"disabled"};
+	size_t const in_sizes[] = {strlen(in_values[0])};
+	int			 n = (int) ARRAY_SIZE(names);
+
+	rc = daos_pool_set_attr(arg->pool.poh, n, names, in_values,
+		in_sizes, NULL);
+	return rc;
+}
+
+/**
+ * Enabled time based Aggrgation for Pool.
+ */
+static int
+enabled_pool_aggregation(void **state, daos_pool_info_t *pinfo)
+{
+	test_arg_t *arg = *state;
+	int rc;
+	char const *const names[] = {"reclaim"};
+	void const *const in_values[] = {"time"};
+	size_t const in_sizes[] = {strlen(in_values[0])};
+	int			 n = (int) ARRAY_SIZE(names);
+
+	rc = daos_pool_set_attr(arg->pool.poh, n, names, in_values,
+		in_sizes, NULL);
+	return rc;
+}
```

Comment:
Minor: these two functions can be made into one to avoid dup code, like set_reclaim_strategy(daos_handle_t poh, const char *strategy), the passed in 'strategy' can be 'disabled', 'time' or 'lazy'.

Reply:
I will made this function into one, I was not sure if we have some more parameter to time in future in case we want to handle it.